code/game/machinery/telecomms/machine_interactions.dm

```diff
@@ -214,6 +214,52 @@
 				change_frequency = 0
 				temp = "<font color = #666633>-% Frequency changing deactivated %-</font color>"
 
+/obj/machinery/telecomms/server/Options_Menu()
+	var/dat= {"</ol>
+			<h2>Frequency Names:</h2>"}
+	if(length(freq_names))
+		dat += "<ul>"
+		for(var/x in freq_names)
+			dat += "<li>"
+			if(freq_names[x])
+				dat += "<font color = [freq_names[x]]>"
+			dat += "[x]"
+			if(freq_names[x])
+				dat += "</font color>"
+			dat += "<a href='?src=\ref[src];delete_name=[x]'>\[X\]</a></li>"
+		dat += "</ul>"
+	else
+		dat += "<li>NONE</li>"
+	
+	dat += {"<p><a href='?src=\ref[src];input_name=1'>\[Add Frequency Name\]</a></p>
+			<hr />"}
+	return dat
+
+/obj/machinery/telecomms/server/Options_Topic(href, href_list)
+
+	if(href_list["delete_name"])
+		var/x = href_list["delete_name"]
+		temp = "<font color = #666633>-% Removed frequency name [x] %-</font color>"
+		freq_names.Remove(x)
+
+	if(href_list["input_name"])
+		var/newfreq = input(usr, "Specify a new frequency name.", src, network) as null|text
+		if(newfreq && canAccess(usr))
+			if(!(newfreq == SYND || newfreq == RAIDER || newfreq == REV_COMM))
+				if(!(newfreq in freq_names))
+					freq_names.Add(newfreq)
+					temp = "<font color = #666633>-% New frequency name assigned: \"[newfreq]\" %-</font color>"
+		
+					var/newcolor = input(usr, "Specify a new frequency color. Leave blank for defaults.", src, network) as null|color
+					if(newcolor && canAccess(usr))
+						freq_names[newfreq] = newcolor
+						temp = "<font color = [newcolor]>-% New frequency color assigned. %-</font color>"
+
+					var/indx = freq_names.Find(newfreq)
+					if(indx && indx <= freq_listening.len && freq_listening[indx])
+						update_radio_frequency(newfreq, freq_listening[indx], newcolor, usr)
+			else
```

Comment:
this else statement appears to want to be 1 further indent to be equivalent to the name assignment test, currently it is given a name denial for a rejected frequency, and there's no feedback for bad freq

Reply:
Attempted to fix this by squashing redundant if()s